wrangler-transform/src/main/java/co/cask/wrangler/Wrangler.java

```diff
@@ -440,6 +443,37 @@ public void transform(StructuredRecord input, Emitter<StructuredRecord> emitter)
     }
   }
 
+  private Object getValue(StructuredRecord input, String fieldName) {
+    Schema fieldSchema = input.getSchema().getField(fieldName).getSchema();
+    fieldSchema = fieldSchema.isNullable() ? fieldSchema.getNonNullable() : fieldSchema;
+    Schema.Type fieldType = fieldSchema.getType();
+
+    switch (fieldType) {
```

Comment:
don't need this switch. Just switch on the logical type, and put the `return input.get(fieldName)` after that.